### PR TITLE
Simplified Position API

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
cc: @pakrym, @joshfree, @davidfowl

### CODE REVIEW

File: src/System.Collections.Sequences/System/Collections/Sequences/Position.cs [hunk 1]
```diff
@@ -16,20 +16,19 @@ namespace System.Collections.Sequences
 
         public static implicit operator Position(int index) => new Position(index, null);
```
Comment (pakrym) [inline src/System.Collections.Sequences/System/Collections/Sequences/Position.cs]: Do we need all these operators, `Get<T>`, and static `Create` method? 
Can we just expose item and index and add a single public constructor that takes them?

Comment (pakrym) [reply src/System.Collections.Sequences/System/Collections/Sequences/Position.cs]: Also in all of the usages of `Get<T>` you never check that item is not null. Do we need to add some kind of validation that position was of an expected type?

Comment (KrzysztofCwalina) [reply src/System.Collections.Sequences/System/Collections/Sequences/Position.cs]: That's how it was originally, but you guys wanted to make it harder for people to get access to the object. Get/Create allow you to store an internal type in object to make it impossible to retrieve by third party code.

Comment (pakrym) [reply src/System.Collections.Sequences/System/Collections/Sequences/Position.cs]: You can always do `Get<object>` and get around it. And `ReadOnlyBuffer` stores publicly available types there. I don't think we can protect it from retrieving the item.

Comment (KrzysztofCwalina) [reply src/System.Collections.Sequences/System/Collections/Sequences/Position.cs]: Yes, but I wanted to change the check to only return the object if typeof(T) == _item.GetType().
I was planing to do it in a subsequent PR.

**Comment by KrzysztofCwalina** [comment]
Let me merge this one. I will play with the typeof(T) == _item.GetType() check, and if it does not work, we can go back to Item and Index properties.

**Pull request merged**